core-plugin/src/main/groovy/com/github/jrubygradle/api/core/AbstractJRubyPrepare.groovy

```diff
@@ -0,0 +1,97 @@
+package com.github.jrubygradle.api.core
+
+import com.github.jrubygradle.api.gems.GemOverwriteAction
+import com.github.jrubygradle.api.gems.GemUtils
+import groovy.transform.CompileStatic
+import org.gradle.api.DefaultTask
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+
+import static com.github.jrubygradle.api.gems.GemOverwriteAction.SKIP
+import static com.github.jrubygradle.api.gems.GemUtils.extractGems
+import static com.github.jrubygradle.api.gems.GemUtils.setupJars
+
+/** Abstract base class for building custom tasks for preparing GEMs.
+ *
+ * @author Schalk W. Cronjé
+ * @author R Tyler Croy
+ * @author Christian Meier
+ *
+ * @since 2.0
+ */
+@CompileStatic
+abstract class AbstractJRubyPrepare extends DefaultTask implements JRubyAwareTask {
```

Comment:
Most of this code has been in the JRubyPrepare task before, but has now been pulled into this new base class. The reason is that the core plugin can be re-used by the Asciidoctot Gradle plugin for instance without exposing any of the functionality in the base plugin